design/sketch.webidl

```diff
@@ -65,6 +65,9 @@ interface WebGPUBufferUsage {
     const u32 VERTEX = 32;
     const u32 UNIFORM = 64;
     const u32 STORAGE = 128;
+    const u32 TEXTURE_BUFFER = 256;
```

Comment:
We didn't get invetigations for texture buffers and indirect parameter buffers. AFAIK they aren't portable but we should double check.

Reply:
Vulkan and D3D12 seem portable in their Indirect parameter buffer structs (same argument order w.r.t. semantics and data type)

Unindexed
https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/VkDrawIndirectCommand.html
https://docs.microsoft.com/en-us/windows/desktop/api/D3D12/ns-d3d12-d3d12_draw_arguments

Indexed
https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/VkDrawIndexedIndirectCommand.html
https://docs.microsoft.com/en-us/windows/desktop/api/D3D12/ns-d3d12-d3d12_draw_indexed_arguments

Compute
https://docs.microsoft.com/en-us/windows/desktop/api/D3D12/ns-d3d12-d3d12_dispatch_arguments
https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/VkDispatchIndirectCommand.html


As for sourcing the draw count in MDI from a buffer, it seems that Vulkan (via VK_KHR_draw_indirect_count) and Metal may be compatible
https://github.com/KhronosGroup/MoltenVK/issues/168

That would also imply that Metal is compatible with the above indirect draw and dispatch parameter buffers.

The only change WebGPU may have to make is add more usages, because D3D12 seems to split the `INDIRECT` Vulkan usage into 3 distinct usages (un-indexed, indexed and dispatch).

Reply:
Also a Khronos discussion about whether to map HLSL's `Buffer<>` to "Uniform Texel Buffer"
https://github.com/KhronosGroup/glslang/issues/795

Reply:
please move this to an investigation issue, like ones we have now: https://github.com/gpuweb/gpuweb/issues?q=is%3Aissue+is%3Aopen+label%3Ainvestigation

Reply:
Also see https://github.com/gpuweb/gpuweb/issues/31 that discusses indirect commands